File: frontend/src/services/learningPath.ts
/**
 * 学习路径服务
 */
import { api } from './api'

export interface LearningPathLesson {
  id: number
  learning_path_id: number
  lesson_id: number
  order_index: number
  is_required: boolean
  created_at: string
  lesson_title?: string
  lesson_description?: string | null
  lesson_cover_image?: string | null
  lesson_difficulty?: string | null
  lesson_rating?: number
  lesson_duration?: number | null
}

export interface LearningPath {
  id: number
  title: string
  description: string | null
  creator_id: number
  difficulty_level: string
  cover_image_url: string | null
  is_published: boolean
  estimated_hours: number | null
  created_at: string
  updated_at: string
}

export interface LearningPathWithLessons extends LearningPath {
  lessons: LearningPathLesson[]
  lesson_count: number
  creator_name: string
}

export interface LearningPathListItem extends LearningPath {
  lesson_count: number
  creator_name: string
}

export interface LearningPathCreate {
  title: string
  description?: string
  difficulty_level?: string
  cover_image_url?: string
  estimated_hours?: number
  lessons?: Array<{
    lesson_id: number
    order_index: number
    is_required?: boolean
  }>
}

export interface LearningPathUpdate {
  title?: string
  description?: string
  difficulty_level?: string
  cover_image_url?: string
  estimated_hours?: number
  is_published?: boolean
}

class LearningPathService {
  /**
   * 创建学习路径（教师/研究员）
   */
  async createLearningPath(data: LearningPathCreate): Promise<LearningPath> {
    return await api.post<LearningPath>('/learning-paths/', data)
  }

  /**
   * 更新学习路径
   */
  async updateLearningPath(pathId: number, data: LearningPathUpdate): Promise<LearningPath> {
    return await api.put<LearningPath>(`/learning-paths/${pathId}`, data)
  }

  /**
   * 删除学习路径
   */
  async deleteLearningPath(pathId: number): Promise<void> {
    await api.delete(`/learning-paths/${pathId}`)
  }

  /**
   * 获取学习路径列表
   */
  async getLearningPaths(publishedOnly: boolean = true): Promise<LearningPathListItem[]> {
    return await api.get<LearningPathListItem[]>('/learning-paths/', {
      params: { published_only: publishedOnly }
    })
  }

  /**
   * 获取学习路径详情
   */
  async getLearningPath(pathId: number): Promise<LearningPathWithLessons> {
    return await api.get<LearningPathWithLessons>(`/learning-paths/${pathId}`)
  }

  /**
   * 向学习路径添加课程
   */
  async addLessonToPath(
    pathId: number,
    lessonId: number,
    orderIndex: number,
    isRequired: boolean = true
  ): Promise<LearningPathLesson> {
    return await api.post<LearningPathLesson>(`/learning-paths/${pathId}/lessons`, {
      lesson_id: lessonId,
      order_index: orderIndex,
      is_required: isRequired
    })
  }

  /**
   * 从学习路径移除课程
   */
  async removeLessonFromPath(pathId: number, lessonId: number): Promise<void> {
    await api.delete(`/learning-paths/${pathId}/lessons/${lessonId}`)
  }
}

export const learningPathService = new LearningPathService()

File: backend/app/schemas/learning_path.py
"""
学习路径相关的Pydantic Schema
"""
from datetime import datetime
from typing import Optional, List
from pydantic import BaseModel


class LearningPathLessonCreate(BaseModel):
    """添加课程到学习路径"""
    lesson_id: int
    order_index: int
    is_required: bool = True


class LearningPathLessonResponse(BaseModel):
    """学习路径课程响应"""
    id: int
    learning_path_id: int
    lesson_id: int
    order_index: int
    is_required: bool
    created_at: datetime
    
    class Config:
        from_attributes = True


class LearningPathLessonWithDetails(LearningPathLessonResponse):
    """带课程详情的学习路径课程"""
    lesson_title: str
    lesson_description: Optional[str]
    lesson_cover_image: Optional[str]
    lesson_difficulty: Optional[str]
    lesson_rating: float
    lesson_duration: Optional[int]


class LearningPathCreate(BaseModel):
    """创建学习路径"""
    title: str
    description: Optional[str] = None
    difficulty_level: str = "beginner"  # beginner, intermediate, advanced
    cover_image_url: Optional[str] = None
    estimated_hours: Optional[int] = None
    lessons: List[LearningPathLessonCreate] = []


class LearningPathUpdate(BaseModel):
    """更新学习路径"""
    title: Optional[str] = None
    description: Optional[str] = None
    difficulty_level: Optional[str] = None
    cover_image_url: Optional[str] = None
    estimated_hours: Optional[int] = None
    is_published: Optional[bool] = None


class LearningPathResponse(BaseModel):
    """学习路径响应"""
    id: int
    title: str
    description: Optional[str]
    creator_id: int
    difficulty_level: str
    cover_image_url: Optional[str]
    is_published: bool
    estimated_hours: Optional[int]
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True


class LearningPathWithLessons(LearningPathResponse):
    """带课程列表的学习路径"""
    lessons: List[LearningPathLessonWithDetails]
    lesson_count: int
    creator_name: str


class LearningPathListItem(LearningPathResponse):
    """学习路径列表项"""
    lesson_count: int
    creator_name: str

